varats/tools/research_tools/vara.py

```diff
@@ -0,0 +1,202 @@
+"""
+Module for the research tool VaRA that describes the VaRA code base layout and
+how to configure and setup VaRA.
+"""
+import typing as tp
+import os
+import logging
+from pathlib import Path
+import shutil
+
+from PyQt5.QtCore import (QProcess)
+
+from plumbum import local
+from plumbum.cmd import mkdir, ln
+
+from varats.settings import CFG, save_config
+from varats.tools.research_tools.research_tool import (ResearchTool, CodeBase,
+                                                       SubProject)
+from varats.vara_manager import (BuildType, run_process_with_output,
+                                 set_vara_cmake_variables, ProcessManager)
+from varats.utils.exceptions import ProcessTerminatedError
+from varats.utils.cli_util import log_without_linsep
+
+LOG = logging.getLogger(__name__)
+
+
+class VaRACodeBase(CodeBase):
+    """
+    Layout of the VaRA code base: setting up vara-llvm-project fork, VaRA, and
+    optinaly phasar, for static analysis.
+    """
+
+    def __init__(self, base_dir: Path) -> None:
+        sub_projects = [
+            SubProject("vara-llvm-project",
+                       "https://github.com/llvm/llvm-project.git", "upstream",
+                       "vara-llvm-project"),
+            SubProject("VaRA", "git@github.com:se-passau/VaRA.git", "origin",
+                       "vara-llvm-project/vara"),
+            SubProject(
+                "phasar",
+                "https://github.com/secure-software-engineering/phasar.git",
+                "origin", "vara-llvm-project/phasar")
+        ]
+        super().__init__(base_dir, sub_projects)
+
+    def setup_vara_remotes(self) -> None:
+        """
+        Sets up VaRA specific upstream remotes for projects that where forked.
+        """
+        self.get_sub_project("vara-llvm-project").add_remote(
+            self.base_dir, "origin",
+            "git@github.com:se-passau/vara-llvm-project.git")
+
+    def setup_build_link(self) -> None:
+        """
+        Setup build-config folder link for VaRAs default build setup scripts.
+        """
+        llvm_project_dir = self.base_dir / self.get_sub_project(
+            "vara-llvm-project").path
+        mkdir(llvm_project_dir / "build/")
+        with local.cwd(llvm_project_dir / "build/"):
+            ln("-s", llvm_project_dir / "vara/utils/vara/builds/", "build_cfg")
+
+    def checkout_vara_version(self, version: int,
+                              use_dev_branches: bool) -> None:
+        """
+        Checkout out a specific version of VaRA.
+
+        Args:
+            version: major version number, e.g., 100 or 110
+            use_dev_branches: true, if one wants the current development version
+        """
+        dev_suffix = "-dev" if use_dev_branches else ""
+        LOG.info(f"Checking out VaRA version {version}" + dev_suffix)
```

Comment:
Would something like this work:

```suggestion
        dev_suffix = "-dev" if use_dev_branches else ""
        LOG.info(f"Checking out VaRA version {str(version) + dev_suffix}")
```

I don't like that mix of fstr and string concatenation.

Reply:
I guess